src/replication.c

```diff
@@ -1731,6 +1731,20 @@ void readSyncBulkPayload(connection *conn) {
          * dictionaries. */
         diskless_load_backup = disklessLoadMakeBackup();
     }
+
+    /* Replica starts to apply data from new master, we must discard the cached
+     * master structure. And before this, make sure there is no master client,
+     * since replica's data would be changed and cached master is exactly wrong
+     * once the master client is created. */
```

Comment:
i'm still not entirely sure what you meant to say, i'm removing the word "exactly" which seems wrong.
but i'm still not sure what you meant to say. it seems obvious we shouldn't have either master or cached master at this stage, but i'm not entirely sure what risk you're referring to.
feel free to try and improve.
```suggestion
    /* Replica starts to apply data from new master, we must discard the cached
     * master structure. And before this, make sure there is no master client,
     * since replica's data would be changed and cached master is  wrong once
     *  the master client is created. */
```

Reply:
>  it seems obvious we shouldn't have either master or cached master at this stage, but i'm not entirely sure what risk you're referring to.

Hi @oranagra  Because I remove `replicationDiscardCachedMaster` in some places, especially in [line 2224](https://github.com/redis/redis/pull/9398/files#diff-955c0fee32b574a1d9adb4540b8ff1564b15db9695d88a8dabc66b3c19c825daL2224), i worry about introducing unexpected cases. The annotations seem not clear,  let me remove this part?
```
And before this, make sure there is no master client, since replica's data would be changed and cached master is  wrong once the master client is created.
```

Reply:
ok, if you feel this part is redundant, remove it, if you think it's an important note, try to make it clearer.